### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by quangphuoc00**
**Utility:**

Enhance the maintainability and enable location tracker to select the best source to get the current location.

**Current State of Development:**

- Add `LocationTracker` and `PermissionUtil` classes.
- Replace `LocationRequestUpdate` with `FusedLocationProviderClient` to select the best source to get the current location.
- Fix Rightmesh checkstyle warnings.

**Outstanding Problems:**
None

**Outstanding Challenges:**
None

### CODE REVIEW

File: app/src/main/java/rightmesh/left/io/gpstracker/LocationTracker.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+import com.google.android.gms.tasks.OnSuccessListener;
+
+/**
+ * Using FusedLocationProviderClient to get location updates.
+ */
+public class LocationTracker implements LifecycleObserver {
+    private FusedLocationProviderClient fusedLocationProviderClient;
+    private Activity activity;
+
+    private int interval = 1000;
```
Comment (FrazerSeymour) [inline app/src/main/java/rightmesh/left/io/gpstracker/LocationTracker.java:36]: Nitpick: These variables `private final static int ALL_CAPS_GLOBALS`.

File: app/src/main/java/rightmesh/left/io/gpstracker/LocationTracker.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+        this.fusedLocationProviderClient = LocationServices
+                .getFusedLocationProviderClient(activity);
+        lifecycle.addObserver(this);
+    }
+
+    /**
+     * Set up the preferred the rate in millisecond to receive updates.
+     *
+     * @param interval in milliseconds
+     * @return {@link LocationTracker}
```
Comment (FrazerSeymour) [inline app/src/main/java/rightmesh/left/io/gpstracker/LocationTracker.java:62]: It is helpful to describe the content that is being returned, beyond just the type. For example this tag you could say

`@return this {@link LocationTracker} instance`.

Elsewhere maybe `@return {@link Type} representing whatever, of <foo> in <this edge case>`.

**Review by compscidr — APPROVED** [review]
Looks good to me.

File: app/src/main/java/rightmesh/left/io/gpstracker/utils/PermissionUtil.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+     * @param grantResults The grant results for the corresponding permissions
+     */
+    public void handleResult(int requestCode,
+                             @NonNull String[] permissions,
+                             @NonNull int[] grantResults) {
+        if (requestCode == PERMISSION_REQUEST_CODE) {
+            ArrayList<String> grantedPermissions = new ArrayList<>();
+            ArrayList<String> deniedPermission = new ArrayList<>();
+
+            for (int i = 0; i < permissions.length; i++) {
```
Comment (FrazerSeymour) [inline app/src/main/java/rightmesh/left/io/gpstracker/utils/PermissionUtil.java:116]: I believe you can do `for (String permission: permissions)` even on primitive arrays, if you don't want to have to deal with indices.

**Review by FrazerSeymour — APPROVED** [review]
Looks great. :+1:

**Pull request merged**